pkg/jx/cmd/install.go

```diff
@@ -381,151 +383,229 @@ func (options *InstallOptions) Run() error {
-			return gitOpsModifyConfigMap(templatesDir, name, nil, configStore, callback)
-		}
-		options.modifySecretCallback = func(name string, callback func(secret *core_v1.Secret) error) (*core_v1.Secret, error) {
-			return gitOpsModifySecret(templatesDir, name, nil, configStore, callback)
-		}
+	options.configureHelm(client, originalNs)
+	err = options.installHelmBinaries()
+	if err != nil {
+		return errors.Wrap(err, "installing helm binaries")
 	}
 
-	if options.Flags.Provider == EKS {
-		var deps []string
-		d := binaryShouldBeInstalled("eksctl")
-		if d != "" {
-			deps = append(deps, d)
-		}
-		d = binaryShouldBeInstalled("heptio-authenticator-aws")
-		if d != "" {
-			deps = append(deps, d)
+	err = options.installCloudProviderDependencies()
+	if err != nil {
+		return errors.Wrap(err, "installing cloud provider dependencies")
+	}
+
+	err = options.configureKubectl(ns)
+	if err != nil {
+		return errors.Wrap(err, "configure the kubectl")
+	}
+
+	options.Flags.Provider, err = options.GetCloudProvider(options.Flags.Provider)
+	if err != nil {
+		return errors.Wrapf(err, "retrieving cloud provider '%s'", options.Flags.Provider)
+	}
+
+	err = options.setMinikubeFromContext()
+	if err != nil {
+		return errors.Wrap(err, "configuring minikube from kubectl context")
+	}
+
+	err = options.configureCloudProivderPreInit(client)
+	if err != nil {
+		return errors.Wrap(err, "configuring the cloud provider before initializing the platform")
+	}
+
+	err = options.init()
+	if err != nil {
+		return errors.Wrap(err, "initializing the Jenkins X platform")
+	}
+
+	err = options.configureCloudProivderPostInit(client, ns)
+	if err != nil {
+		return errors.Wrap(err, "configuring the cloud provider after initializing the platform")
+	}
+
+	err = options.createSystemVault(client, ns)
+	if err != nil {
+		return errors.Wrap(err, "creating the system vault")
+	}
+
+	err = options.configureDockerRegistry(client, ns)
+	if err != nil {
+		return errors.Wrap(err, "configuring the docker registry")
+	}
+
+	err = options.configureHelmValues(ns)
+	if err != nil {
+		return errors.Wrap(err, "configuring helm values")
+	}
+
+	err = options.configureGitAuth(originalGitUsername, originalGitServer, originalGitToken)
+	if err != nil {
+		return errors.Wrap(err, "configuring the git auth")
+	}
+
+	cloudEnvDir, err := options.cloneJXCloudEnvironmentsRepo()
+	if err != nil {
+		return errors.Wrap(err, "cloning the jx cloud environments repo")
+	}
+
+	if options.Flags.Provider == "" {
+		return fmt.Errorf("no Kubernetes provider found to match cloud-environment with")
+	}
+	providerEnvDir := filepath.Join(cloudEnvDir, fmt.Sprintf("env-%s", strings.ToLower(options.Flags.Provider)))
+	valuesFiles, secretsFiles, temporaryFiles, err := options.getHelmValuesFiles(configStore, providerEnvDir)
+	if err != nil {
+		return errors.Wrap(err, "getting the helm value files")
+	}
+
+	log.Infof("Installing Jenkins X platform helm chart from: %s\n", providerEnvDir)
+
+	options.Verbose = true
```

Comment:
This seems odd. Shouldn't it be up to the user to specify whether to be Verbose or not? (or is it required to have output from subsequent methods? - if so, i think this could do with a comment as to _why_ it is being done)

Reply:
This was original in the code in this place. I haven't tired to remove it. I think we should remove it and let the use to set the `Verbose` options? @jstrachan What do you think?

Reply:
a while back someone tried to make the commands not verbose; but when installing when things go wrong all kinds of useful error message from tools like git, helm, gcloud et al get gobbled up without verbose. 

So we should default verbose to true for `jx install / jx create cluster ...` for sure